e2e/test-component/scenarios/embedding-sdk/interactive-question-theming.cy.spec.tsx

```diff
@@ -132,31 +118,117 @@ describe(
 
         // Should be the slightly darker version of the background color, same as the notebook button hover
         cy.get(customColumn).should(($el) =>
-          haveBackgroundColor($el, darken(theme.colors?.background, 0.05)),
+          assertBackgroundColorEqual($el, darken(BACKGROUND_COLOR, 0.05)),
         );
 
         // Hover should be an even darker version of the background color
         cy.get(customColumn)
           .should("be.visible")
           .realHover()
           .should(($el) =>
-            haveBackgroundColor($el, darken(theme.colors?.background, 0.1)),
+            assertBackgroundColorEqual($el, darken(BACKGROUND_COLOR, 0.1)),
           );
       });
     });
+
+    it("table cell color should follow the background color", () => {
+      const BACKGROUND_COLOR = "rgb(200, 210, 220)";
+
+      setupInteractiveQuestionWithTheme({
+        colors: { background: BACKGROUND_COLOR },
+      });
+
+      getSdkRoot().within(() => {
+        cy.findByTestId("table-header").should(($el) => {
+          assertBackgroundColorEqual($el, BACKGROUND_COLOR);
+        });
+
+        cy.findByTestId("table-body")
+          .findAllByRole("gridcell")
+          .first()
+          .should(($el) => {
+            assertBackgroundColorEqual($el, BACKGROUND_COLOR);
+          });
+      });
+    });
+
+    it("table.cell.backgroundColor should override table cell color", () => {
+      const CELL_COLOR = "rgb(123, 111, 222)";
+
+      setupInteractiveQuestionWithTheme({
+        colors: { background: "rgb(200, 210, 220)" },
+        components: {
+          table: { cell: { backgroundColor: CELL_COLOR } },
+        },
+      });
+
+      getSdkRoot().within(() => {
+        cy.findByTestId("table-header").should(($el) => {
+          assertBackgroundColorEqual($el, CELL_COLOR);
+        });
+
+        cy.findByTestId("table-body")
+          .findAllByRole("gridcell")
+          .first()
+          .should(($el) => {
+            assertBackgroundColorEqual($el, CELL_COLOR);
+          });
+      });
+    });
+
+    it("getColorDifferencePercentage compares colors correctly", () => {
```

Comment:
This is used by our `assertBackgroundColorEqual` assertion directly. Basically, the `lighten(...)` and `darken(...)` method we use has a one-off problem, so `#2DD2EE` can be `#2DD2EF` when read from the browser. This results in a 0.25% difference in RGB color which we can detect easily.